### ISSUE TITLE
SignalR keep reconnecting by itself after connected  and recoonect for 120 sec

### ISSUE THREAD

**Issue by sattha** [SignalR/java-client#61]
LOG CAT
=========
01-22 15:37:50.102    7939-8795/com.example.SignalRRR I/System.out﹕ stateChanged From [Connected] To [Reconnecting]
01-22 15:39:32.932    7939-8905/com.example.SignalRRR I/System.out﹕ stateChanged From [Reconnecting] To [Connected]
01-22 15:39:35.162    7939-8952/com.example.SignalRRR I/System.out﹕ stateChanged From [Connected] To [Reconnecting]
01-22 15:41:23.122    7939-9052/com.example.SignalRRR I/System.out﹕ stateChanged From [Reconnecting] To [Connected]
01-22 15:41:25.352    7939-9359/com.example.SignalRRR I/System.out﹕ stateChanged From [Connected] To [Reconnecting]
01-22 15:43:13.642    7939-9478/com.example.SignalRRR I/System.out﹕ stateChanged From [Reconnecting] To [Connected]
01-22 15:43:15.872    7939-9481/com.example.SignalRRR I/System.out﹕ stateChanged From [Connected] To [Reconnecting]
01-22 15:45:13.652    7939-9501/com.example.SignalRRR I/System.out﹕ stateChanged From [Reconnecting] To [Connected]
01-22 15:45:15.892    7939-9505/com.example.SignalRRR I/System.out﹕ stateChanged From [Connected] To [Reconnecting]

Here is my code snippet implement for connected

 @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main);
        System.setProperty("http.keepAlive", "true");

        final HubConnection connection = createHubConnection();
        final ClientTransport transport = createTransport(TransportType.LongPolling);
        final HubProxy proxy = connection.createHubProxy("MyHub");

        try {
            connection.start(transport).get();
        } catch (InterruptedException e) {
            e.printStackTrace();
        } catch (ExecutionException e) {
            e.printStackTrace();
        }
    }

**Comment by arnoldpistorius** [comment]
I think this is normal behaviour for LongPolling transport.

**Comment by anzaidemirzoi** [comment]
then what to do in reconnecting? how to connect again ?

**Comment by arnoldpistorius** [comment]
My English is not very well, but I hope you understand;

When you use the LongPolling transport, each time the server sends data to the client, the connection will be closed and an new connection will be started. Because the server only responds when the server needs to send data or on timeouts (120s) the connection state stays in reconnecting. This is what LongPolling is. LongPolling connections are just simple http(s) connections (which don't support push messaging from server to client). So the request is sent (reconnecting) and waits for the response or timeout (connected).

**Comment by anzaidemirzoi** [comment]
Yes I know this, thanks though :) but the problem is after first reconnecting state, it never connects again.

**Comment by anzaidemirzoi** [comment]
maybe you can send us a proper signalR android implementation for LongPollingTransport ?

**Comment by sattha** [comment]
I solve my own problem by let android ping to SignalR server before it reconnect.
I'm not sure this is a normal practice or not but it keep my android client connection alive without long reconnect problem.

However, I still don't know why it reconnect so quickly. 
I can't make it wait for 120s as document said.
So, that why I have to ping to SignalR server every 10s
to prevent magic reconnect behavior (which take a long time).

my implementation post in here...
http://stackoverflow.com/a/28131378/4208710

**Comment by anzaidemirzoi** [comment]
ping solves my problem. thank you very much

**Comment by MinChanSike** [comment]
You can handle connection reconnect immediately.

 mConnection.closed(new Runnable() {
            @Override
            public void run() {
                Log.d(LOG_TAG, "DISCONNECTED!");
                if (mConnection.getState().toString() == "Disconnected") {
                    new reconnectPulling().execute();
                }
            }
        });

 private class reconnectPulling extends AsyncTask<Void, Void, Void> {

        @Override
        protected Void doInBackground(Void... params) {
            for (int i = 0; i < 5; i++) {
                try {
                    if (mConnection.getState().toString() != "Connected") {
                        mConnection.start();
                        Log.d(LOG_TAG, "Reconnect To Server...");
                        Thread.sleep(5000);
                    }
                } catch (InterruptedException e) {
                    Thread.interrupted();
                }
            }
            return null;
        }
    }

**Comment by MinChanSike** [comment]
You can handle connection reconnect immediately.
```
 mConnection.connected(new Runnable() {
            @Override
            public void run() {
                Log.d(LOG_TAG, "CONNECTED.");
            }
        });
 mConnection.closed(new Runnable() {
            @Override
            public void run() {
                Log.d(LOG_TAG, "DISCONNECTED!");
                if (mConnection.getState().toString() == "Disconnected") {
                    new reconnectPulling().execute();
                }
            }
        });
```
```
private class reconnectPulling extends AsyncTask<Void, Void, Void> {

        @Override
        protected Void doInBackground(Void... params) {
            for (int i = 0; i < 5; i++) {
                try {
                    if (mConnection.getState().toString() != "Connected") {
                        mConnection.start();
                        Log.d(LOG_TAG, "Reconnect To Server...");
                        Thread.sleep(5000);
                    }
                } catch (InterruptedException e) {
                    Thread.interrupted();
                }

            }
            return null;
        }
    }

```

**Comment by sattha** [comment]
This is a problem at server side configuration
Ping approach that I mention before is not a real fixed !

Please see this.. 
http://www.asp.net/signalr/overview/guide-to-the-api/handling-connection-lifetime-events#connectiontimeout

However, even look at document it still quite confusing to me.
Here is my last working code (server side).
For **LongPolling** this will hold connection for 1 min. 
and continue polling by itself immediately, no delay (if no other issue) 

```
protected void Application_Start(object sender, EventArgs e)
{
    // Make long polling connections wait a maximum of 110 seconds for a
    // response. When that time expires, trigger a timeout command and
    // make the client reconnect.
    GlobalHost.Configuration.ConnectionTimeout = TimeSpan.FromSeconds(60);
    
    // Wait a maximum of 30 seconds after a transport connection is lost
    // before raising the Disconnected event to terminate the SignalR connection.
    GlobalHost.Configuration.DisconnectTimeout = TimeSpan.FromSeconds(180);
    
    // For transports other than long polling, send a keepalive packet every
    // 10 seconds. 
    // This value must be no more than 1/3 of the DisconnectTimeout value.
    GlobalHost.Configuration.KeepAlive = TimeSpan.FromSeconds(60);
}
```

Ps. For some reason, whether document said keepAlive value is auto disable for LongPolling
but it still have effect to connection also.

**Comment by aspnet-hello** [comment]
All issues in this repo are being closed because this repo is no longer in use. Please see the readme for more information: https://github.com/SignalR/java-client/blob/master/readme.md.
